ibis/backends/tests/test_string.py

```diff
@@ -663,60 +676,13 @@ def uses_java_re(t):
                 ),
             ],
         ),
-        param(
-            lambda t: t.date_string_col.split("/"),
-            lambda t: t.date_string_col.str.split("/"),
-            id="split",
-            marks=pytest.mark.notimpl(
-                [
-                    "impala",
-                    "mysql",
-                    "sqlite",
-                    "mssql",
-                    "druid",
-                    "oracle",
-                    "flink",
-                    "exasol",
-                ],
-                raises=com.OperationNotDefinedError,
-            ),
-        ),
-        param(
-            lambda t: ibis.literal("-").join(["a", t.string_col, "c"]),
-            lambda t: "a-" + t.string_col + "-c",
-            id="join",
-            marks=pytest.mark.notimpl(
-                ["exasol"],
-                raises=com.OperationNotDefinedError,
-            ),
-        ),
-        param(
-            lambda t: t.string_col + t.date_string_col,
-            lambda t: t.string_col + t.date_string_col,
-            id="concat_columns",
-        ),
-        param(
-            lambda t: t.string_col + "a",
-            lambda t: t.string_col + "a",
-            id="concat_column_scalar",
-        ),
-        param(
-            lambda t: "a" + t.string_col,
-            lambda t: "a" + t.string_col,
-            id="concat_scalar_column",
-        ),
-        param(
-            lambda t: t.string_col.replace("1", "42"),
-            lambda t: t.string_col.str.replace("1", "42"),
-            id="replace",
-        ),
     ],
 )
-def test_string(backend, alltypes, df, result_func, expected_func):
-    expr = result_func(alltypes).name("tmp")
+def test_substring(backend, alltypes, df, result_func, expected_func):
```

Comment:
the diff for these test cases is really messy, sorry. What I did:
- move some test cases to separate test_substring test
- remove from xfails/brokens since I fixed/implemented things
- removed no test cases
- added the `positive-index` test case

Reply:
Thanks for the explanation, I was going to ask why you deleted test cases 😂